jupyter/4 - dbGap FHIR Controlled Access via RAS -  COPDGene.ipynb

```diff
@@ -0,0 +1,1058 @@
+       "28  phs002409                            FHIR Test Study's ALPHA"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "sdict = []\n",
+    "for s in studies:\n",
+    "    sdict.append({'id':s['id'], 'title':s['title']})\n",
+    "\n",
+    "\n",
+    "import pandas as pd\n",
+    "\n",
+    "df = pd.DataFrame.from_dict(sdict)\n",
+    "df.sort_values(by=['id'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3c61c42e",
+   "metadata": {},
+   "source": [
+    "🖐 If of interest obtain two different TSTs and run the same query and add steps here compare the results you get."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b79428d4",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6eb2d137",
+   "metadata": {},
+   "source": [
+    "### Look at COPDGene study details\n",
+    "\n",
+    "Returning to the COPDGene study phs000179"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b6b2e7d",
+   "metadata": {},
+   "source": [
+    "### Retrieving Observations for Subjects in the study\n",
+    "We take the same approach as in notebook 3.\n",
+    "\n",
+    "Because we have 10368 subjects, in the interests of time the for loop steps through every 100th subject. \n",
+    "This will still take several minutes to retrieve all the results.\n",
+    "\n",
+    "Note: more efficient ways of retrieiving the data are being explored. Feedback from users of these notebooks is encouraged to drive what features of the API would be useful. The slower methods here are useful for now as they allow us all to develop familiarity with the data, and to understand what API features would be most useful to work with the data. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "7ac54d92",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "5b8215d54c894ee98fbf41664941dbaa",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "IntProgress(value=0, max=10368)"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "104 patients had observations\n",
+      "25085 total observations\n"
+     ]
+    }
+   ],
+   "source": [
+    "from ipywidgets import IntProgress\n",
+    "from IPython.display import display\n",
+    "\n",
+    "prog = IntProgress(min=0, max=len(patients)) # instantiate the bar\n",
+    "display(prog) # display the bar\n",
+    "\n",
+    "increment = 100\n",
+    "\n",
+    "all_obs = []\n",
+    "patients_with_obs = []\n",
+    "for p in patients[10:10368:increment]:\n",
+    "    obs = mf.run_query(f\"Observation?subject={p['id']}\", show_stats=False)\n",
```

Comment:
The FHIR way to do this query (which we do not currently support) is:
```
https://dbgap-api.ncbi.nlm.nih.gov/fhir/x1/Observation?subject._has:ResearchSubject:individual:study=phs002410
```

Reply:
Going with what we currently support. Also working on an approach that only downloads Observations of interest - which we do support